### ISSUE TITLE
Grid Switcher quits after hotkey pressed

### ISSUE THREAD

**Issue by circulon** [cspotcode/virtual-desktop-grid-switcher#17]
Hi 

I am frustrated with this as it seems to be an ideal app and the autohotkey integration is a really nice touch.

Unfortunately it seems broken ;(
I have followed the installation instructions and done the following
installed Autohotkey current (1.1.24.01) 
made sure to unblock the zip and extracted it to a folder in my home dir
added a shortcut for the autohotkeys/ .akk file to my start items
I have setup 6 desktops through the task view and adjusted the VDGS settings to 3 cols and 2 rows.

This works as expected and on login I can see the VDGS icon and the AHK script icon in the taskbar

So as noted in the docs I do ctrl+win+right arrow ... nothing happens but a few seconds later I see the VDGS icon disappear.
I choose reload script for the AHK icon and the VDGS restarts.
use task view to move to a different desktop and repeat as above and VDGS quits ;(

If I suspend hotkeys for the script ctrl+win+right or left arrows behave as expected and the VDGS does not quit.

Is there logs or other info that I can provide to help figure this out?

Thanks

**Comment by cspotcode** [comment]
What version of Windows 10 are you running?  Have you received the "anniversary update" yet?  Windows tweaked their APIs a bit for the anniversary update and I haven't yet updated this project accordingly.  Therefore you'll have to wait for an update to use this project with the Anniversary Update.  Specifically, pulling in updates to a third-party dependency, [Grabacr07/VirtualDesktop](https://github.com/Grabacr07/VirtualDesktop), should fix the problem. (see #16)

This is a hobby project for me so I can't say when it'll be updated.  Feel free to send a pull request if you're able.  Alternatively, "Watch" issue #16 so you'll get a notification when it's resolved.

**Comment by circulon** [comment]
New to Windows 10 ;) and have no Idea what the anniversary update is?
Running on my work PC
Windows 10 Enterprise 
Version : 1607
Build : 14393.51

As I have no idea on how to compile this it looks like I will have to wait

Thanks for the info though

**Comment by cspotcode** [comment]
Yup, that looks like the Anniversary Update.  It was basically a big service pack for Windows 10 released about a month ago.  If you're curious you can Google it.

Please click "watch" on issue #16.  That way you'll be notified when I release a version that works.

**Comment by cspotcode** [comment]
Closed as duplicate of #16
